pandas/io/tests/json/test_json_table_schema.py

```diff
@@ -0,0 +1,159 @@
+"""Tests for Table Schema integration."""
+import json
+from collections import OrderedDict
+
+import numpy as np
+import pandas as pd
+
+from pandas import DataFrame
+import pandas.util.testing as tm
+from pandas.io.json import (
+    as_json_table_type, build_table_schema
+)
+
+
+class TestJSONTableSchema(tm.TestCase):
+
+    def setUp(self):
+        self.df = DataFrame(
+            {'A': [1, 2, 3, 4],
+             'B': ['a', 'b', 'c', 'c'],
+             'C': pd.date_range('2016-01-01', freq='d', periods=4),
+             'D': pd.timedelta_range('1H', periods=4, freq='T'),
```

Comment:
test with other dtypes (e.g. assert that they come up with the correct mappings (or raising is ok if its totally not supported)), IOW
categorical of strings, datetime w/tz, period , floats, different dtypes for integer & float.